### ISSUE TITLE
TypeError: HTTPConnection.request() got an unexpected keyword argument 'chunked'

### ISSUE THREAD

**Issue by liudonghua123** [lukapeschke/dockerfile-from-image#2]
I tried to use this tool to get some info of a specified image, but it failed with `` TypeError: HTTPConnection.request() got an unexpected keyword argument 'chunked' ``.

<details><summary>Details</summary>

```
[root@localhost dockerfile-from-image]# docker run --rm -v '/var/run/docker.sock:/var/run/docker.sock' lukapeschke/dfa eolinker/cluster-redis:6.2.7                                                                                         
Traceback (most recent call last):
  File "/root/entrypoint.py", line 54, in <module>
    my_obj = MainObj()
             ^^^^^^^^^
  File "/root/entrypoint.py", line 15, in __init__
    self._get_image(argv[-1])
  File "/root/entrypoint.py", line 26, in _get_image
    imgs = self.cli.images()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/api/image.py", line 40, in images
    res = self._result(self._get(self._url("/images/json"), params=params),
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/utils/decorators.py", line 47, in inner
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/client.py", line 139, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 790, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 496, in _make_request
    conn.request(
TypeError: HTTPConnection.request() got an unexpected keyword argument 'chunked'
[root@localhost dockerfile-from-image]#
[root@localhost dockerfile-from-image]# docker images | grep eolinker
eolinker/apinto-gateway                   latest                       8c8d9de1ac9f   3 weeks ago      535MB
eolinker/apinto-dashboard                 latest                       1f474e272ce2   5 weeks ago      273MB
eolinker/cluster-redis                    6.2.7                        4cfebc77974f   12 months ago    113MB
[root@localhost dockerfile-from-image]# docker run --rm -v '/var/run/docker.sock:/var/run/docker.sock' lukapeschke/dfa 4cfebc77974f
Traceback (most recent call last):
  File "/root/entrypoint.py", line 54, in <module>
    my_obj = MainObj()
             ^^^^^^^^^
  File "/root/entrypoint.py", line 15, in __init__
    self._get_image(argv[-1])
  File "/root/entrypoint.py", line 26, in _get_image
    imgs = self.cli.images()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/api/image.py", line 40, in images
    res = self._result(self._get(self._url("/images/json"), params=params),
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/utils/decorators.py", line 47, in inner
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/docker/client.py", line 139, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 790, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/urllib3/connectionpool.py", line 496, in _make_request
    conn.request(
TypeError: HTTPConnection.request() got an unexpected keyword argument 'chunked'
[root@localhost dockerfile-from-image]#
```

</details>

**Comment by liudonghua123** [comment]
I found `docker-py` is very old, maybe not compatible with the latest new version of docker, I can use [`docker`](https://github.com/docker/docker-py/) packages to do the similar jobs.

I will create a new dockerfile-from-image tool in my spare time later.

**Comment by liudonghua123** [comment]
Thanks for your original code, I changed a little code and it works for me now. 

See https://github.com/liudonghua123/dockerfile-from-image.

Anyone who interested could tried with `docker run --rm -v '/var/run/docker.sock:/var/run/docker.sock' liudonghua123/dockerfile-from-image <IMAGE_ID_OR_NAME>`

**Comment by lukapeschke** [comment]
Ah yes the image might deserve an update... If you'd like to create a PR here to apply the updates you did, I'd be happy to review and merge it :)

**Comment by liudonghua123** [comment]
@lukapeschke Hi, I adjusted the code, and created a PR, see https://github.com/lukapeschke/dockerfile-from-image/pull/3.

**Comment by lukapeschke** [comment]
Thanks! Left a review there

**Comment by lukapeschke** [comment]
Fixed by #3 , thanks a lot @liudonghua123

**Comment by liudonghua123** [comment]
You are welcome. 😃
